src/main/java/io/nats/client/PublishOptions.java

```diff
@@ -234,17 +245,27 @@ public Builder expectedLastSubjectSequence(long sequence) {
          * Sets the message id. Message IDs are used for de-duplication
          * and should be unique to each message payload.
          * @param msgId the unique message id.
-         * @return builder
+         * @return The Builder
          */
         public Builder messageId(String msgId) {
             this.msgId = emptyAsNull(msgId);
             return this;
         }
 
+        /**
+         * Sets the TTL for this specific message to be published
+         * @param msgTtlSeconds the ttl in seconds
+         * @return The Builder
+         */
+        public Builder msgTtlSeconds(int msgTtlSeconds) {

```

Comment:
`Nats-TTL` also supports the string `"never"`, to indicate the message will never be removed, even if it would otherwise expire due to the stream's `MaxAge`.